src/java/edu/slu/tpen/servlet/ProjectServlet.java

```diff
@@ -91,65 +101,87 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
-                    Project proj = new Project(projID);
-                    //System.out.println("Project 3");
-                    if (proj.getProjectID() > 0) {
-                        //System.out.println("Project 4");
-                        if (new Group(proj.getGroupID()).isMember(uid) || skip) {
-                           // System.out.println("export");
-                            if (checkModified(req, proj)) {
-                               // System.out.println("Project 5");
-                                resp.setContentType("application/ld+json; charset=UTF-8");
-                                //resp.setHeader("Etag", req.getContextPath() + "/manifest/"+proj.getProjectID());
-                                //We expect projects to change, often.  Let's just help quick page refresh.
-                                try{
-                                    String lastModifiedDate = proj.getModification().toString().replace(" ", "T");
-                                    //Note that dates like 2021-05-26T10:39:19.328 have been rounded to 2021-05-26T10:39:19.328 in browser headers.  Account for that here.
-                                    if(lastModifiedDate.contains(".")){
-                                        //If-Modified-Since and Last-Modified headers are rounded.  Wed, 26 May 2021 10:39:19.629 GMT becomes Wed, 26 May 2021 10:39:19 GMT.
-                                        lastModifiedDate = lastModifiedDate.split("\\.")[0];
-                                    }
-                                    LocalDateTime ldt = LocalDateTime.parse(lastModifiedDate, DateTimeFormatter.ISO_DATE_TIME);
-                                    ZonedDateTime fromObject = ldt.atZone(ZoneId.of("GMT"));
-                                    resp.setHeader("Last-Modified", fromObject.format(DateTimeFormatter.RFC_1123_DATE_TIME));
-                                }
-                                catch(DateTimeParseException ex){
-                                    System.out.println("Last-Modified Header could not be formed.  Bad date value for project "+proj.getProjectID());
-                                }
-                                catch(Exception e){
-                                
-                                }
-                                resp.setHeader("Access-Control-Allow-Headers", "*");
-                                resp.setHeader("Access-Control-Expose-Headers", "*"); //Headers are restricted, unless you explicitly expose them.  Darn Browsers.
-                                resp.setHeader("Cache-Control", "no-cache, no-store, must-revalidate"); // HTTP 1.1.
-                                resp.setHeader("Pragma", "no-cache"); // HTTP 1.0.
-                                resp.setHeader("Expires", "0"); // Proxies.
+                    String[] urlData = req.getPathInfo().substring(1).split("/");
```

Comment:
This is technically nicer if you split on the expected pattern "project/" and "manifest/" but the servlet mapping should be maintaining this so it isn't required. You will only find odd behavior with something like "../manifest/3065/193881" which is wrong.